Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs

```diff
@@ -5,119 +5,129 @@
 
 using System;
 using System.Collections.Generic;
-using System.Data.Entity;
 using System.Linq;
-using EdFi.Common;
-using EdFi.Common.Utils.Extensions;
-using EdFi.Security.DataAccess.Contexts;
 using EdFi.Security.DataAccess.Models;
+using Action = EdFi.Security.DataAccess.Models.Action;
 
 namespace EdFi.Security.DataAccess.Repositories
 {
-    public class SecurityRepository : SecurityRepositoryBase, ISecurityRepository
+    /// <summary>
+    /// Provides higher level access to raw security metadata.
+    /// </summary>
+    public class SecurityRepository : ISecurityRepository
     {
-        private readonly ISecurityContextFactory _securityContextFactory;
+        private readonly ISecurityTableGateway _securityTableGateway;
 
-        public SecurityRepository(ISecurityContextFactory securityContextFactory)
+        public SecurityRepository(ISecurityTableGateway securityTableGateway)
         {
-            _securityContextFactory = Preconditions.ThrowIfNull(securityContextFactory, nameof(securityContextFactory));
-
-            Initialize(
-                GetApplication,
-                GetActions,
-                GetClaimSets,
-                GetResourceClaims,
-                GetAuthorizationStrategies,
-                GetClaimSetResourceClaimActions,
-                GetResourceClaimActionAuthorizations);
+            _securityTableGateway = securityTableGateway ?? throw new ArgumentNullException(nameof(securityTableGateway));
         }
 
-        private Application GetApplication()
+        public virtual Action GetActionByName(string actionName)
         {
-            using var context = _securityContextFactory.CreateContext();
-
-            return context.Applications.First(
-                app => app.ApplicationName.Equals("Ed-Fi ODS API", StringComparison.InvariantCultureIgnoreCase));
+            return _securityTableGateway.GetActions().First(a => a.ActionName.Equals(actionName, StringComparison.OrdinalIgnoreCase));
```

Comment:
The hard-coded "Ed-Fi ODS API" looks strange here. I don't like it. But we need to keep it, because other applications that depend on this library expect that.